content/tutorials/aws-persistence.textile

```diff
@@ -0,0 +1,249 @@
+---
+authors:
+- author_bio: ''
+  author_image: 
+  author_name: Steven Applebly
+  author_profile_url: 
+category:
+- integrations
+date_published: '2016-07-22T02:38:02+01:00'
+excerpt: Learn how to persist messages into AWS S3.
+group: sdk
+index: 10
+languages:
+- javascript
+last_updated: '2021-11-19T16:35:45+00:00'
+level: easy
+platform: mixed
+reading_time: 5 — 10
+section: tutorials
+tags:
+- Integrations
+- AWS Lambda
+title: AWS Persistence in S3
+meta_description: "How to persist data into S3."
+meta_keywords: "Ably realtime, AWS Lambda Functions, integrations"
+---
+
+Ably "webhooks":/general/events enable users to configure custom rules that react to channel events. For example, triggering an action when a message is sent, or when a user joins the channel. You can use our integration with "AWS Lambda":https://aws.amazon.com/lambda/ to create these rules.
+
+In this tutorial you will learn how to create a channel rule that intercepts messages sent to a channel and persists them to a file in "AWS Simple Storage Service (S3)":https://aws.amazon.com/s3/.
```

Comment:
I think we could explain the context a bit more here. I think we should also have an overview diagram here too. See the [Kinesis tutorial](https://ably.com/tutorials/reactor-firehose-kinesis) for an example.

I think it's also worth adding a note to the effect that this approach is fine for low volume message rates, but S3 isn't really optimized for high volumes of small messages, which is what typically flows through Ably. @lmars recommended some form of batching of messages into, for example, 1MB chunks before persisting. We don't have to implement that in this tutorial, but we could suggest it.

Also, I didn't see a link to our [AWS Authentication](https://ably.com/docs/general/aws-authentication) page. I think it would be very useful to link to it fairly early on, before the user goes into the details, as it could provide some useful background reading if they are relatively new to AWS.